go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/aws/amazon-cloudwatch-agent-operator
 
-go 1.20
```

Comment:
Does changing the go version have any side-effects?

Reply:
AFAIK no, its backward compatible with 1.20 according to the release notes - https://tip.golang.org/doc/go1.21 so it should be fine